cmake/partition_manager.cmake

```diff
@@ -437,8 +450,66 @@ else()
             )
 
           # There is no padding in front of the network core application.
-          math(EXPR net_app_TO_SECONDARY
-            "${PM_MCUBOOT_SECONDARY_ADDRESS} - ${net_app_addr} + ${PM_MCUBOOT_PAD_SIZE}")
+	  if(CONFIG_PM_EXTERNAL_FLASH)
+	    if(CONFIG_SOC_NRF5340_CPUAPP)
+	      set(EXTERNAL_FLASH_START_ADDRESS 0x10000000)
+            else()
+	      set(EXTERNAL_FLASH_START_ADDRESS 0x12000000)
+            endif()
+            math(EXPR net_app_TO_SECONDARY "${EXTERNAL_FLASH_START_ADDRESS} + ${CONFIG_PM_EXTERNAL_FLASH_BASE}
```

Comment:
All of these calculations looks complicated.

I would suggest extracting this to a separate file and adding comments to all the calculations explaining why they are needed.

Reply:
All those calculations looks similar, is it not possible to just make a single function or macro with inputs, something like:
```
# secondary_slot([EXT_START <address>] [EXT_BASE <address>])
function(secondary_slot [EXT_START <address>] [EXT_BASE <address>] )
  cmake_parse_arguments(SEC_SLOT "" "EXT_START;EXT_BASE" "" ${ARGN}
  get_shared(no IMAGE mcuboot PROPERTY UPDATEABLE_IMAGE_NUMBER)

  if(${no} GREATER 1)
    set(sec_slot_idx "_1")
  endif()

  get_target_property(
    net_app_addr
    partition_manager
    CPUNET_PM_APP_ADDRESS
  )
  
  math(EXPR net_app_to_secondary
    "${SEC_SLOT_EXT_START} + ${SEC_SLOT_EXT_BASE} + ${PM_MCUBOOT_SECONDARY${sec_slot_idx}_ADDRESS} - ${net_app_addr} + ${PM_MCUBOOT${sec_slot_idx}_PAD_SIZE}")
  
  set_property(
    TARGET partition_manager
    PROPERTY net_app_TO_SECONDARY
    ${net_app_TO_SECONDARY}
  )  
endfunction()
``` 

and then change all the if'es to:
```
if(CONFIG_PM_EXTERNAL_FLASH)
  secondary_slot(EXT_START ${EXTERNAL_FLASH_START_ADDRESS} EXT_BASE ${CONFIG_PM_EXTERNAL_FLASH_BASE})
else()
  secondary_slot()
endif()
```
as `UPDATEABLE_IMAGE_NUMBER` is fetched inside the function.
Might even be improved further, and obtain better naming.